### PR TITLE
Save transactions

### PR DESCRIPTION
**Pull request by k3ntako**

### CODE REVIEW

**Comment by k3ntako** [comment]
[Splitwise - Save transaction (5pt)](https://trello.com/c/C4ONLG2j/53-splitwise-save-transaction-5pt)

File: src/Prompter.ts [hunk 1]
```diff
@@ -6,22 +6,26 @@ export interface IPrompter {
   userIO: IUserIO;
   promptList(message: string, choices: (string | Separator)[]): Promise<Answers>;
   promptInput(message: string): Promise<Answers>;
+  promptConfirm(message: string): Promise<Answers>;
+  promptNumber(message: string): Promise<Answers>;
+  promptDate(message: string): Promise<Answers>;
```
Comment (olgaboiar) [inline src/Prompter.ts:11]: is `promptDate` asking for an input that is specifically date? if yes, what kind of input `promptInput` is asking for?

File: src/Prompter.ts [hunk 2]
```diff
@@ -41,4 +45,35 @@ export default class Prompter implements IPrompter{
 
     return await this.userIO.prompt(options);
   }
+
+  async promptConfirm(message: string): Promise<Answers> {
```
Comment (olgaboiar) [inline src/Prompter.ts:49]: looks a lot like `promptList`, doesn't it? and `promptNumber`..

File: src/pages/AddTransactionPage.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      throw Error('The other user cannot be undefined.')
+    }
+
+    const name: string = await this.getName();
+    const date: Date = await this.getDate();
+    const userPaid: boolean = await this.getUserPaid();
+    const cost: number = await this.getCost();
+
+    const lender: IUser = userPaid ? this.user : otherUser;
+    const borrower: IUser = userPaid ? otherUser : this.user;
```
Comment (olgaboiar) [inline src/pages/AddTransactionPage.ts:35]: is it possible to check `userPaid` just once? might it be the case to use destructuring assignment? (I'm not sure, just wondering)

File: src/pages/AddTransactionPage.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
+import IPage from './IPage';
+import MenuPage from './MenuPage';
+import { IPrompter } from '../Prompter';
+import { IUserIO } from '../CLI';
+import { Answers } from 'inquirer';
+import { userTable, transactionTable } from '../tables';
+import { IUser } from '../tables/UserTable';
+import chalk from 'chalk';
+
+export default class AddTransactionPage implements IPage {
```
Comment (olgaboiar) [inline src/pages/AddTransactionPage.ts:10]: Think about what this page does, and if something violates the single responsibility principle

File: src/pages/MenuPage.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import AddTransactionPage from './AddTransactionPage';
+import { IUser } from '../tables/UserTable';
+import Separator from 'inquirer/lib/objects/separator';
+
+interface INextPageOptions {
+  [key: string]: new (userIO: IUserIO, prompter: IPrompter, user: IUser) => IPage;
+  'Add transaction': new (userIO: IUserIO, prompter: IPrompter, user: IUser) => IPage;
+}
+
+export default class MenuPage implements IPage {
```
Comment (olgaboiar) [inline src/pages/MenuPage.ts:14]: do you think all this pages have a lot of things in common?

File: src/tables/TransactionTable.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    if (cost <= 0) {
+      throw new Error('Expected cost to be a positive number')
+    }
+
+    if ((cost * 100) % 1) {
+      throw new Error('Cost cannot go past the hundredths')
+    }
+  }
+
+  private halveCost(cost: number): number {
```
Comment (olgaboiar) [inline src/tables/TransactionTable.ts:62]: `splitCost`? :)

File: src/Postgres.ts [hunk 7]
```diff
@@ -66,22 +83,52 @@ export default class Postgres implements IPostgres {
...
 
   async getAllUsers(): Promise<IUser[]> {
-    const result: QueryResult = await this.pool.query('SELECT * FROM users;');
+    const result: QueryResult = await this.query('SELECT * FROM users;');
 
     return result.rows;
   }
 
+  async createTransaction(name: string, date: Date, cost: number): Promise<ITransaction> {
+    const dateStr = `${date.getFullYear()}-${date.getMonth() + 1}-${date.getDate()}`;
```
[the remark this answers is not in the document]
Comment (k3ntako) [reply src/Postgres.ts:103]: The SQL query is a string, so that is why I am converting it to a string. Does that make sense?

**Pull request merged**